File: 0005-longest-palindromic-substring/0005-longest-palindromic-substring.cs
public class Solution
{
    public string LongestPalindrome(string s)
    {
        int n = s.Length;
        if (n == 0) return "";
        bool[,] dp = new bool[n, n];
        int maxLength = 1;
        int start = 0;

        for (int i = 0; i < n; i++)
            dp[i, i] = true;

        for (int i = 0; i < n - 1; i++)
        {
            if (s[i] == s[i + 1])
            {
                dp[i, i + 1] = true;
                start = i;
                maxLength = 2;
            }
        }

        for (int len = 3; len <= n; len++)
        {
            for (int i = 0; i <= n - len; i++)
            {
                int j = i + len - 1;
                if (s[i] == s[j] && dp[i + 1, j - 1])
                {
                    dp[i, j] = true;
                    if (len > maxLength)
                    {
                        start = i;
                        maxLength = len;
                    }
                }
            }
        }

        return s.Substring(start, maxLength);
    }
}

File: 0005-longest-palindromic-substring/0005-longest-palindromic-substring.cpp
class Solution {
public:
    string longestPalindrome(string s) {
        if (s.empty()) return "";
        int len = s.size();
        int start = 0;
        int end = 0;
        for(int i=0;i<len;i++){
            int len1 =checkLongestsub(s,i,i);
            int len2 =checkLongestsub(s,i,i+1);
            int maxLen = max(len1,len2);
            if(maxLen>(end - start)){
                start = i - (maxLen - 1)/2;
                end = i+maxLen/2;

            }

        }
        return s.substr(start,end - start + 1);    
    }
    int checkLongestsub(string s ,int b,int e){
        int len = s.size(); 
        while(b>=0 && e<len && s[b]==s[e]){
            b--;
            e++;
        }
        return e-b-1;
    }
};